backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -378,22 +379,71 @@ boolean childContains(Set<?> list, Set<?> sublist) {
 
   @Override
   public ComputeRepositoryDiffRequest.Response computeRepositoryDiff(
-      RepositoryFunction repositoryFunction, ComputeRepositoryDiffRequest request)
-      throws ModelDBException {
+      RepositoryDAO repositoryDAO, ComputeRepositoryDiffRequest request) throws ModelDBException {
     try (Session session = ModelDBHibernateUtil.getSessionFactory().openSession()) {
       session.beginTransaction();
-      RepositoryEntity repositoryEntity = repositoryFunction.apply(session);
 
-      CommitEntity internalCommitA = session.get(CommitEntity.class, request.getCommitA());
+      boolean invalidParam = false;
+      if (!request.getBranchA().isEmpty()
+          && !request.getBranchB().isEmpty()
+          && !request.getCommitA().isEmpty()
+          && !request.getCommitB().isEmpty()) {
+        invalidParam = true;
+      } else if (!request.getBranchA().isEmpty() && !request.getCommitA().isEmpty()) {
+        invalidParam = true;
+      } else if (request.getBranchA().isEmpty() && request.getCommitA().isEmpty()) {
+        invalidParam = true;
+      } else if (!request.getBranchB().isEmpty() && !request.getCommitB().isEmpty()) {
+        invalidParam = true;
+      } else if (request.getBranchB().isEmpty() && request.getCommitB().isEmpty()) {
+        invalidParam = true;
+      }
```

Comment:
```suggestion
      final boolean invalidParam;
      if (request.getBranchA().isEmpty()
          && request.getBranchB().isEmpty()
          && !request.getCommitA().isEmpty()
          && !request.getCommitB().isEmpty()
          || !request.getBranchA().isEmpty()
          && !request.getBranchB().isEmpty()
          && request.getCommitA().isEmpty()
          && request.getCommitB().isEmpty()) {
        invalidParam = false;
      } else  {
        invalidParam = true;
      }
```
maybe this variant is simpler?

Reply:
oh, sorry misunderstood the logic

Reply:
just ignore this comment